### ISSUE TITLE
[native-filters]Disable Apply button when new filter selected in a required filter(initial state) 

### ISSUE THREAD

**Issue by junlincc** [apache/superset#15004]
Expected behavior: 
When user check the Required* checkbox in the modal, and there's no filter value selected in that required filter, Apply button should be disabled, and the required field should be highlighted, in red to draw user attention. 

Current behavior: When user check the Required* checkbox in the modal, and there's no filter value selected in that required filter, Apply button works and pass a strange WHERE clause to the query even no filter value is selected. 

<img width="1499" alt="Screen Shot 2021-06-04 at 6 22 29 PM" src="https://user-images.githubusercontent.com/67837651/120875677-de7a7900-c561-11eb-99e0-dcc1c49b473d.png">

 
https://user-images.githubusercontent.com/67837651/120875495-f8678c00-c560-11eb-8410-a2faebaa1e1e.mov

**Comment by junlincc** [comment]
@villebro the team mentioned this is expected behavior which puzzled me a bit. 😵 I understand this is to prevent undesired expensive query, can we simply disable the Apply button when the required field has no value?

**Comment by villebro** [comment]
An alternative (and more intuitive solution) could be the following:
- On the modal: make it impossible to save a filter config when the "Required" flag is set unless a default value is defined
- on the filter tab: disable the apply button if any filter has "required" set but doesn't have a value

**Comment by junlincc** [comment]
agreed with both. #2 is probably more straight forward. @villebro please go ahead with the proposed solution. 🙏

**Comment by jinghua-qa** [comment]
Verified fixed in 2021.34
